privval/remote_signer.go

```diff
@@ -15,8 +17,9 @@ import (
 // RemoteSignerClient implements PrivValidator, it uses a socket to request signatures
 // from an external process.
 type RemoteSignerClient struct {
-	conn net.Conn
-	lock sync.Mutex
+	conn            net.Conn
+	consensusPubKey crypto.PubKey
+	lock            sync.Mutex
```

Comment:
`mtx` for consistency

Reply:
do we need it anyways? removed it from the FilePV in https://github.com/tendermint/tendermint/pull/2870 - afaik, it's not actually necessary since we don't access it concurrently

Reply:
The justification seems to be this: https://github.com/tendermint/tendermint/pull/2568#discussion_r223306805
and AFAIU this can't happen; the sign method would never be called with the pub key package as the amino prefix identifies the right message types. Removed.

Reply:
Hmm, maybe I'm wrong and the mtx is still necessary: https://circleci.com/gh/tendermint/tendermint/39026?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
The prefix there (A6160A51) is not any of the registered messages (RemoteSignerMsg).

Maybe this is a new variant of https://github.com/tendermint/tendermint/issues/2677#issuecomment-433938289 which doesn't deadlock & timeout anymore.

Reply:
It actually looks like with the mutex in place, we sometimes deadlock (in tests, haven't seen this in happening in the wild and I can't reproduce it locally) and without it we seem to get messages out of order or mixed up as Hendrik described in above comment: (https://github.com/tendermint/tendermint/pull/2568#discussion_r223306805)
Keeping the mutex seems more robust but I think we further need to investigate what is wrong and we probably do not need it (and need to update the tests instead).